src/DynamoCore/Models/PresetsState.cs

```diff
@@ -0,0 +1,113 @@
+using System.Xml;
+using Dynamo.Nodes;
+
+namespace Dynamo.Models
+{
+    /// <summary>
+    /// This class references a set of nodemodels, and a set of serialized versions of those nodemodels
+    /// a client can use this class to store the state of a set of nodes from a graph
+    /// </summary>
+    public class PresetState
+    {
+        private Guid guid;
+        private readonly List<NodeModel> nodes;
+        private readonly List<XmlElement> serializedNodes;
+
+        # region properties
+
+        public string Name { get; private set; }
+        public string Description { get; private set; }
+       
+        /// <summary>
+        /// list of nodemodels that this state serializes
+        /// </summary>
+        public IEnumerable<NodeModel> Nodes { get{return nodes;}}
+
+        /// <summary>
+        /// list of serialized nodes
+        /// </summary>
+        public IEnumerable<XmlElement> SerializedNodes { get { return serializedNodes; }}
+
+        /// <summary>
+        /// A unique identifier for the state.
+        /// </summary>
+        public Guid Guid
+        {
+            get { return guid; }
+        }
+
+        #endregion
+
+        #region constructor
+        /// <summary>
+        /// create a new presetsState, this will serialize all the referenced nodes by calling their serialize method, 
+        /// the resulting XML elements will be used to save this state when the presetModel is saved on workspace save
+        /// the below temp root and doc is to avoid exceptions thrown by the zero touch serialization methods
+        /// </summary>
+        /// <param name="name">name for the state, must not be null </param>
+        /// <param name="description">description of the state, can be null</param>
+        /// <param name="inputsToSave">set of nodeModels, must not be null</param>
+        /// <param name="id">an id GUID, can be empty GUID</param>
+        public PresetState(string name, string description, IEnumerable<NodeModel> inputsToSave, Guid id)
+        {
+            if (String.IsNullOrEmpty(name))
+            {
+                throw new ArgumentNullException("name");
+            }
+
+            if (inputsToSave == null || inputsToSave.Count() < 1)
+            {
+                throw new ArgumentNullException("inputsToSave");
+            } 
+
+            //if we have not supplied a guid at construction then create a new one
+            if (id == Guid.Empty)
+            {
+                guid = Guid.NewGuid();
+            }
+            else
+            {
+                guid = id;
+            }
+            
+            Name = name;
+            Description = description;
+            nodes = inputsToSave.ToList();
+
+           
+            var tempdoc = new XmlDocument();
+            serializedNodes = new List<XmlElement>();
+            foreach (var node in Nodes)
+            {
+                serializedNodes.Add(node.Serialize(tempdoc, SaveContext.Preset));
```

Comment:
@pboyer 
instead of modifying zero touch base function which explicitly checks that document path is not null on file save, I introduced a new save context for presets which avoid the save context.file path in zero touch nodes. I think this is more clear and keeps the zero touch serialization failing where it should.